infra/README.md

```diff
@@ -0,0 +1,130 @@
+This is a work in progress to use Terraform to manage our cloud
+infrastructure.
+
+Instructions:
+
+- You will need a GCP project.  Configure `gcloud` to point at your project:
+
+   ```
+   gcloud config set project <gcp-project-id>
+   gcloud config set compute/zone <gcp-zone>
+   ```
+
+- Create a service account for Terraform with Owner role, create a
+  service account key and place it in
+  `$HOME/.hail/terraform_sa_key.json`.
+
+- Enable the the GCP services needed by Hail:
+
+   ```
+   gcloud services enable \
+       compute.googleapis.com \
+       cloudresourcemanager.googleapis.com \
+       servicenetworking.googleapis.com \
+       sqladmin.googleapis.com \
+       container.googleapis.com \
+       serviceusage.googleapis.com
+   ```
+
+- Install terraform.
+
+- Create `$HOME/.hail/global.tfvars` that looks like:
+
+   ```
+   gcp_project = "<gcp-project-id>"
+   gcp_region = "<gcp-region>"
+   gcp_zone = "<gcp-zone>"
+   domain = "<domain>"
+   ```
+
+- Run `terraform init`.
+
+- Run `terraform apply -var-file="$HOME/.hail/global.tfvars"`.
+  Terraform has created a GKE cluster named `vdc`.  Configure `kubectl` to point at the vdc cluster:
+
+  ```
+  gcloud container clusters get-credentials vdc
+  ```
+
+- Go to the Google Cloud console, VPC networks > internal > Private
+  service connection > Private connections to services, and enable
+  Export custom routes to both connections.
+
+You can now install Hail:
+
+- Create a VM on the internal network, 100GB, Ubuntu 20.04 TLS, allow
+  full access to all Cloud APIs.  10GB will run out of space.  We
+  assume the rest of the commands are run on the VM.
```

Comment:
This pretty much worked for everything except the $HAIL/letsencrypt step, due to a permissions error. I ended up running that outside the VM, but it probably wouldn't be too difficult to fix:

```
Error from server (Forbidden): error when creating "service.yaml": roles.rbac.authorization.k8s.io is forbidden: User "<redacted>" cannot create resource "roles" in API group "rbac.authorization.k8s.io" in the namespace "default": requires one of ["container.roles.create"] permission(s).
Error from server (Forbidden): error when creating "service.yaml": rolebindings.rbac.authorization.k8s.io is forbidden: User "<redacted>" cannot create resource "rolebindings" in API group "rbac.authorization.k8s.io" in the namespace "default": requires one of ["container.roleBindings.create"] permission(s).
make: *** [Makefile:30: run] Error 1
```

Reply:
This works when using e.g. the terraform service account for running the VM.

Also it seems good to use at least 32 GB of RAM for the VM (otherwise compiling the query service didn't work for me).

Reply:
Great, I now recommend a n2-standard-8, and to run the VM as a the terraform service account.